### DISCUSSION TITLE
Update of metadata entity object record in ES/OS index doesn't remove deleted fields.

### DISCUSSION THREAD

**Discussion by ramzanzan** [open-metadata/OpenMetadata#16889] [General]
Hello dear OpenMetadata developers! 

I have found that while update of any metadata entity (such as Table) reindexing of record in ES/OS doesn't remove fields which removed in original entity (such as owner).

I have found that problem is in ES script for update [(link to src)](https://github.com/open-metadata/OpenMetadata/blob/96417452a946e8b6ae085fc5bdb5382bc1c40c93/openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java#L38). This script doesn't count enitity's removed fields cause those fields doesn't present in script params -> there aren't `calls ctx._source.put(k, params.get(k))` for them.

Is such behavior a bug?
Why do you use scritptedUpdate and not just upsert?

**Comment by mohityadav766** [comment]
@ramzanzan if you remove the owner , is it not updating then?

**Reply by ramzanzan** [reply]
Yes. It will be removed from the relational database but will remain in the index in ES

**Reply by ramzanzan** [reply]
@mohityadav766  So, can you please clarify what is purpose of use of scripted update ([link to src)](https://github.com/open-metadata/OpenMetadata/blob/96417452a946e8b6ae085fc5bdb5382bc1c40c93/openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java#L38) instead of just update?

**Reply by harshach** [reply]
@ramzanzan what you mean removed from relational database. How are you removing it from the UI/API or directly changing it in db

**Reply by ramzanzan** [reply]
From UI/API
